x/stakeibc/keeper/unbonding_records_get_host_zone_unbondings_msgs_test.go

```diff
@@ -1,226 +1,813 @@
+		{
+			ValidatorAddress:   "valA",
+			CurrentDelegation:  sdkmath.NewInt(50),
+			BalancedDelegation: sdkmath.NewInt(0),
+			Capacity:           sdkmath.NewInt(50),
+		},
+		{
+			ValidatorAddress:   "valB",
+			CurrentDelegation:  sdkmath.NewInt(200),
+			BalancedDelegation: sdkmath.NewInt(5),
+			Capacity:           sdkmath.NewInt(195),
+		},
+		{
+			ValidatorAddress:   "valC",
+			CurrentDelegation:  sdkmath.NewInt(1089),
+			BalancedDelegation: sdkmath.NewInt(1000),
+			Capacity:           sdkmath.NewInt(89),
+		},
+	}
 
-	totalWgt := sdk.NewDec(int64(tc.totalWgt))
-	actualAmtToUnbondDec := sdk.NewDecFromInt(actualAmtToUnbond)
-	actualUnbondMsg1 := actualUnbondMsgs[0].String()
-	actualUnbondMsg2 := actualUnbondMsgs[1].String()
-	val1Fraction := sdk.NewDec(int64(tc.hostZone.Validators[0].Weight)).Quo(totalWgt)
-	val2Fraction := sdk.NewDec(int64(tc.hostZone.Validators[1].Weight)).Quo(totalWgt)
-	val1UnbondAmt := val1Fraction.Mul(actualAmtToUnbondDec).TruncateInt().String()
-	val2UnbondAmt := val2Fraction.Mul(actualAmtToUnbondDec).TruncateInt().String()
+	// Build list of input validators and map of balanced delegations from expected list
+	validators := []*types.Validator{}
+	balancedDelegations := map[string]sdkmath.Int{}
+	for _, validatorCapacity := range expectedUnbondCapacity {
+		validators = append(validators, &types.Validator{
+			Address:    validatorCapacity.ValidatorAddress,
+			Delegation: validatorCapacity.CurrentDelegation,
+		})
+		balancedDelegations[validatorCapacity.ValidatorAddress] = validatorCapacity.BalancedDelegation
+	}
 
-	val1Unbonded := strings.Contains(actualUnbondMsg1, val1UnbondAmt)
-	val2Unbonded := strings.Contains(actualUnbondMsg2, val2UnbondAmt)
+	// Add validators with no capacity - none of these should be in the returned list
+	deficits := []int64{0, 10, 50}
+	valAddresses := []string{"valD", "valE", "valF"}
+	for i, deficit := range deficits {
+		address := valAddresses[i]
+
+		// the delegation amount is arbitrary here
+		// all that mattesr is that it's less than the balance delegation
+		currentDelegation := sdkmath.NewInt(50)
+		balancedDelegation := currentDelegation.Add(sdkmath.NewInt(deficit))
+
+		validators = append(validators, &types.Validator{
+			Address:    address,
+			Delegation: currentDelegation,
+		})
+		balancedDelegations[address] = balancedDelegation
+	}
 
-	// there's rounding in the logic that distributes stake amongst validators, so one or the other of the balances will be correct, depending on the rounding
-	// at least one will be correct, and the other will be off by 1 by rounding, so we check and OR condition
-	s.Require().True(val1Unbonded || val2Unbonded, "unbonding amt should be the correct amount")
+	// Check capacity matches expectations
+	actualUnbondCapacity := s.App.StakeibcKeeper.GetValidatorUnbondCapacity(s.Ctx, validators, balancedDelegations)
+	s.Require().Len(actualUnbondCapacity, len(expectedUnbondCapacity), "number of expected unbondings")
```

Comment:
This is a weird way to do it, why not match the ones below and make it more readable with this:
s.Require().Equal(len(actual...), len(expected...), "message")

Reply:
why is it more unreadable? I would have thought the opposite cause the require statement is clearly indicating that it is checking the length

Reply:
This is definitely weird and unreadable, not your fault it is weird the testing library had a function like this.
Len(actual..., len(expected), "message")    <<< so it looks like a call where the input is all 3 things
but it is doing a comparison between the first and second argument, the first argument is a list, the second argument is a number.